core/src/main/java/ai/timefold/solver/core/impl/domain/variable/declarative/DefaultShadowVariableSessionFactory.java

```diff
@@ -46,46 +46,72 @@ public DefaultShadowVariableSessionFactory(
         this.graphCreator = graphCreator;
     }
 
-    public static <Solution_> VariableReferenceGraph buildGraph(
+    public record GraphDescriptor<Solution_>(ConsistencyTracker<Solution_> consistencyTracker,
```

Comment:
I like this! Although I'd have moved it to a separate file. This one is already long enough to digest in one piece.